sdk/src/Core/Amazon.Runtime/Pipeline/HttpHandler/_mobile/HttpRequestMessageFactory.cs

```diff
@@ -69,54 +68,17 @@ public IHttpRequest<HttpContent> CreateHttpRequest(Uri requestUri)
                 {
                     if (!CanClientConfigBeSerialized(_clientConfig))
                     {
-                        _httpClientCacheRWLock.EnterWriteLock();
-                        try
-                        {
-                            if (_httpClientCache == null)
-                            {
-                                _httpClientCache = CreateHttpClientCache(_clientConfig);
-                            }
-                        }
-                        finally
-                        {
-                            _httpClientCacheRWLock.ExitWriteLock();
-                        }
+                        // atomic operation - if _httpClientCache still equals null, swap in the CreateHttpClientCache 
+                        Interlocked.CompareExchange(ref _httpClientCache, CreateHttpClientCache(_clientConfig), null);
```

Comment:
Does anything need to be done with the created value if the exchange fails as it's already set, or is it OK to just be left to be GC'd?

Reply:
I was assuming that it's safe to just GC it. If not, then we need `Lazy<T>`

Reply:
It's just I can't see what `CreateHttpClientCache()` does from the diff, hence the question.